modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt

```diff
@@ -1,197 +1,138 @@
-        }
-    }
+            val selection = MirrordConfigDialog(
+                "Change mirrord active configuration",
+                configs.keys.toList().sorted()
+            ).show() ?: return
 
-    @Deprecated(
-        "Deprecated in Java",
-        ReplaceWith(
-            "createPopupActionGroup(button, DataManager.getInstance().getDataContext(button))",
-            "com.intellij.ide.DataManager"
-        )
-    )
-    override fun createPopupActionGroup(button: JComponent): DefaultActionGroup {
-        return createPopupActionGroup(button, DataManager.getInstance().getDataContext(button))
+            service.activeConfig = selection.option?.let { configs[it] }
+        }
     }
 
-    private fun getReadablePath(path: String, project: Project): String {
-        val basePath = project.basePath ?: throw Error("couldn't resolve project path")
-        val relativePath = Path.of(basePath).relativize(Path.of(path))
-        return relativePath.toString()
-    }
+    private class SettingsAction : AnAction("Settings") {
+        override fun actionPerformed(e: AnActionEvent) {
+            val service = e.project?.service<MirrordProjectService>() ?: return
 
-    // mostly what is viewed by the user is done here, goal is to have minimal logic in the function
-    // because it is called every half a second
-    override fun update(e: AnActionEvent) {
-        e.project?.let { project ->
-            // this check ensures that we don't query the index when it is being built
-            // querying the index during the startup/Dumb Mode can give us 0 or stale data
-            if (!::configFiles.isInitialized) {
-                e.presentation.isVisible = false
-                blockAndQueryIndex(project)
-                return
+            val configs: MutableMap<String, () -> Unit> = mutableMapOf()
+            service.activeConfig?.let {
+                configs["(active) %s".format(it.presentableUrl)] = {
+                    FileEditorManager.getInstance(service.project).openFile(it, true)
+                }
             }
 
-            blockAndQueryIndex(project)
-
-            if (configFiles.size > 1) {
-                if (chosenFile !in configFiles) {
-                    chosenFile = configFiles.first()
+            val defaultConfig = service.configApi.getDefaultConfig()
+            if (defaultConfig == null) {
+                configs["(create default)"] = {
+                    val config = service.configApi.createDefaultConfig()
+                    FileEditorManager.getInstance(service.project).openFile(config, true)
                 }
-
-                e.presentation.text = chosenFile?.let { getReadablePath(it, project) }
             } else {
-                chosenFile = configFiles.firstOrNull()
+                configs["(default) %s".format(defaultConfig.presentableUrl)] = {
+                    FileEditorManager.getInstance(service.project).openFile(defaultConfig, true)
+                }
             }
```

Comment:
so the Settings always shows active and default only?

Reply:
Yup, at least for now